flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskTest.java

```diff
@@ -1300,9 +1304,7 @@ public TestInvokableCorrect(Environment environment) {
         public void invoke() {}
 
         @Override
-        public void cancel() {
-            fail("This should not be called");
```

Comment:
nit: although this won't cause test failure, I think logging (unexpected) cancellation makes debugging and reading the code a bit easier

Reply:
For sure, it is true but right now we write to log that the cancellation was unexpected while it was actually expected which leads to confusion. It's why I explicitly added a check that the cancellation don't happen in tests where we don't expect it.